devices/Hdc1080/Hdc1080.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Device.I2c;
+using System.Device.Model;
+using UnitsNet;
+
+namespace Iot.Device.Hdc1080
+{
+    /// <summary>
+    /// Temperature and Humidity sensor Hdc1080
+    /// </summary>
+    [Interface("Hdc1080 - temperature and humidity sensor")]
+    public class Hdc1080 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private int _wakeUpTimeMilliseconds = 20;
+
+        /// <summary>
+        /// Default I2C address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x40;
+
+        /// <summary>
+        /// Constructs Hdc1080 instance with high temperature and humidity sampling. Measurement mode set by default to humidity and temperature.
+        /// </summary>
+        /// <param name="i2cDevice">I2C device used to communicate with the device</param>
+        /// <param name="humidityResolution">Humidity sampling resolution</param>
+        /// <param name="temperatureResolution">Temperature sampling resolution</param>
+        public Hdc1080(I2cDevice i2cDevice, HumidityResolution humidityResolution = HumidityResolution.High, TemperatureResolution temperatureResolution = TemperatureResolution.High)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // According to data sheet sensor needs at least 15ms after wake up to be ready.
+            Thread.Sleep(_wakeUpTimeMilliseconds);
+            SetTemperatureResolution(temperatureResolution);
+            SetHumidityResolution(humidityResolution);
+        }
+
+        /// <summary>
+        /// Reads temperature, check data sheet, page 14, 8.6.1 section
+        /// </summary>
+        /// <returns>
+        /// Temperature
+        /// </returns>
+        [Telemetry("Temperature")]
+        public Temperature ReadTemperature() => Temperature.FromDegreesCelsius((ReadData(RegisterAddress.TemperatureMeasurement) * 165 / 65536f) - 40);
+
+        /// <summary>
+        /// Reads relative humidity, check data sheet, page 14, 8.6.2 section
+        /// </summary>
+        /// <returns>
+        /// Relative humidity
+        /// </returns>
+        [Telemetry("Humidity")]
+        public RelativeHumidity ReadHumidity() => RelativeHumidity.FromPercent(ReadData(RegisterAddress.HumidityMeasurement) * 100 / 65536);
+
+        /// <summary>
+        /// Reads device id, check data sheet, page 14, 8.6.6 section
+        /// </summary>
+        /// <returns>
+        /// Device id
+        /// </returns>
+        public string ReadDeviceId() => ReadData(RegisterAddress.DeviceId).ToString();
```

Comment:
Looking at those few ones, it's a typical property. It won't change over time. so something like this for the next 3 ones:
```suggestion
        public string DeviceId { get => ReadData(RegisterAddress.DeviceId).ToString(); }
```